overview/authentication-flows.mdx

```diff
@@ -0,0 +1,195 @@
+    const config = {
+      clientId: "your_client_id",
+      // PKCE enabled by default, no client secret needed
+      redirectUrl: "https://yourapp.com/callback"
+    }
+    ```
+  </Tab>
+
+  <Tab title="Client Secret Only">
+    **Best for:** Traditional server-side applications requiring OAuth 2.0 compatibility.
+
+    ### How it Works
+    - Uses traditional OAuth 2.0 authorization code flow
+    - Client authenticates using client secret
+    - PKCE disabled for compatibility with legacy systems
+    - Suitable for confidential clients with secure storage
+
+    ### Security Benefits
+    - ✅ Familiar OAuth 2.0 authentication pattern
+    - ✅ Client authentication via shared secret
+    - ✅ Compatible with existing OAuth infrastructure
+    - ✅ Server-side credential protection
+
+    ### Configuration
+    ```javascript
+    const config = {
+      clientId: "your_client_id",
+      clientSecret: "your_client_secret", // Contact Civic for this
+      pkce: false, // Disable PKCE for traditional OAuth flow
+      redirectUrl: "https://yourapp.com/callback"
+    }
+    ```
+  </Tab>
+
+  <Tab title="PKCE + Client Secret">
+    **Best for:** Maximum security for confidential clients that can support both methods.
+
+    ### How it Works
+    - Combines PKCE with client secret authentication
+    - Provides maximum security through dual verification
+    - Suitable for high-security server-side applications
+    - Both PKCE challenge and client secret verified
+
+    ### Security Benefits
+    - ✅ **Maximum security** through dual authentication
+    - ✅ Protection against authorization code interception
+    - ✅ Client authentication via secret
+    - ✅ Dynamic PKCE verification per request
+
+    ### Configuration
+    ```javascript
+    const config = {
+      clientId: "your_client_id",
+      clientSecret: "your_client_secret", // Contact Civic for this
+      pkce: true, // Enable PKCE alongside client secret
+      redirectUrl: "https://yourapp.com/callback"
+    }
+    ```
+  </Tab>
+</Tabs>
+
+## When to Use Each Method
+
+<CardGroup cols={3}>
+  <Card title="PKCE Only" icon="browser">
+    - Single-page applications (SPAs)
+    - Mobile applications
+    - Frontend applications (React, Vue, Angular)
+    - When client secrets cannot be securely stored
+    - Default for public clients
+  </Card>
+  <Card title="Client Secret Only" icon="server">
+    - Legacy OAuth 2.0 integrations
+    - Traditional server-side applications
+    - When PKCE is not supported
+    - Simple backend services
+    - Existing OAuth infrastructure
+  </Card>
+  <Card title="PKCE + Client Secret" icon="shield">
+    - **High-security server applications**
+    - Financial or healthcare systems
+    - Enterprise backend services
+    - **Maximum security requirements**
+    - Modern confidential clients
+  </Card>
+</CardGroup>
+
+## Getting Client Secrets
+
+<Info>
+If you need a client secret, contact [support@civic.com](mailto:support@civic.com).
```

Comment:
do we have a mechanism for providing these securely?

Reply:
That's a very good point. We could provide a one-time link and make sure it is single use? What are your thoughts?

Reply:
I would prefer we use the UI I have built, or some variant of it, rather than build new infrastructure for this. But it is fundamentally a product question.